### ISSUE TITLE
[MNG-7634] Revert MNG-5982 and MNG-7417

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8818]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[MNG-7634](https://issues.apache.org/jira/browse/MNG-7634?redirect=false)** and commented

As layed out in MNG-7563, both issues though solve one problem, but cause a regression where direct user input manipulates dependency POMs in a way which is not clear/documented as such. As long as we cannot properly understand and document it, we must revert those changes.

---

**Issue Links:**
- [MNG-7563](https://issues.apache.org/jira/browse/MNG-7563) REGRESSION: User properties now override model properties in dependencies
 (_**"fixes"**_)
- [MNG-7635](https://issues.apache.org/jira/browse/MNG-7635) Investigate how to safely propagate user properties to dependencies (see MNG-5982 and MNG-7417)
 (_**"causes"**_)
- [MNG-7417](https://issues.apache.org/jira/browse/MNG-7417) Several classes do not set properties properly for building requests
 (_**"supercedes"**_)
- [MNG-5982](https://issues.apache.org/jira/browse/MNG-5982) The POM for ... is invalid, transitive dependencies ... while property was overriden
 (_**"supercedes"**_)

**Remote Links:**
- [GitHub Pull Request #916
](https://github.com/apache/maven/pull/916)
- [GitHub Pull Request #917
](https://github.com/apache/maven/pull/917)

**Backported to:** [4.0.0-alpha-4](https://github.com/apache/maven/milestone/3?closed=1), [3.9.0](https://github.com/apache/maven/milestone/2?closed=1), [3.8.7](https://github.com/apache/maven/milestone/108?closed=1)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Strange, it did not fail for me. Let me check.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[sjaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sjaranowski), one of the reasons we **must** fix the user/system properties chaos.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Side note, I consider that most of these are incorrectly used:

```
^[[Aosipovmi@deblndw011x:~/var/Projekte/mit (master *=)
$ grep -r  setSystemProperty core-it-suite/src/test/java/
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4428FollowHttpRedirectTest.java:            verifier.setSystemProperty( "javax.net.ssl.trustStore", storePath );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4428FollowHttpRedirectTest.java:            verifier.setSystemProperty( "javax.net.ssl.trustStorePassword", storePwd );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng0848SystemPropOverridesDefaultValueTest.java:        verifier.setSystemProperty( "config.aliasDefaultExpressionParam", "PASSED" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng0786ProfileAwareReactorTest.java:        verifier.setSystemProperty( "expression.outputFile", "target/expression.properties" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng0786ProfileAwareReactorTest.java:        verifier.setSystemProperty( "activate", "anything" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2136ActiveByDefaultProfileTest.java:        verifier.setSystemProperty( "expression.outputFile", new File( testDir, "target/expression.properties" ).getPath() );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2136ActiveByDefaultProfileTest.java:        verifier.setSystemProperty( "expression.expressions", "project/properties" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2305MultipleProxiesTest.java:            verifier.setSystemProperty( "javax.net.ssl.trustStore", storePath );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2305MultipleProxiesTest.java:            verifier.setSystemProperty( "javax.net.ssl.trustStorePassword", storePwd );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2305MultipleProxiesTest.java:            verifier.setSystemProperty( "maven.artifact.threads", "1" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng2577SettingsXmlInterpolationTest.java:        verifier.setSystemProperty( "usr.MNGIT", "usr-prop-test" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4936EventSpyTest.java:        verifier.setSystemProperty( "maven.ext.class.path", "spy-0.1.jar" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3004ReactorFailureBehaviorMultithreadedTest.java:        verifier.setSystemProperty( "maven.threads.experimental", "1" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3004ReactorFailureBehaviorMultithreadedTest.java:        verifier.setSystemProperty( "maven.threads.experimental", "1" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3004ReactorFailureBehaviorMultithreadedTest.java:        verifier.setSystemProperty( "maven.threads.experimental", "1" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3004ReactorFailureBehaviorMultithreadedTest.java:        verifier.setSystemProperty( "maven.threads.experimental", "2" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3004ReactorFailureBehaviorMultithreadedTest.java:        verifier.setSystemProperty( "maven.threads.experimental", "2" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng3379ParallelArtifactDownloadsTest.java:        verifier.setSystemProperty( "maven.artifact.threads", "16" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng0377PluginLookupFromPrefixTest.java:        verifier.setSystemProperty( "touch.outputFile", "target/file.txt" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4452ResolutionOfSnapshotWithClassifierTest.java:        verifier.setSystemProperty( "mng4452.type", "jar" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4452ResolutionOfSnapshotWithClassifierTest.java:        verifier.setSystemProperty( "mng4452.classifier", "unix" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4452ResolutionOfSnapshotWithClassifierTest.java:        verifier.setSystemProperty( "mng4452.type", "jar" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4452ResolutionOfSnapshotWithClassifierTest.java:        verifier.setSystemProperty( "mng4452.classifier", "win" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4452ResolutionOfSnapshotWithClassifierTest.java:        verifier.setSystemProperty( "mng4452.type", "war" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4452ResolutionOfSnapshotWithClassifierTest.java:        verifier.setSystemProperty( "mng4452.classifier", "win" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4469AuthenticatedDeploymentToCustomRepoTest.java:        verifier.setSystemProperty( "file", "settings.xml" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4469AuthenticatedDeploymentToCustomRepoTest.java:        verifier.setSystemProperty( "groupId", "org.apache.maven.its.mng4469" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4469AuthenticatedDeploymentToCustomRepoTest.java:        verifier.setSystemProperty( "artifactId", "it" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4469AuthenticatedDeploymentToCustomRepoTest.java:        verifier.setSystemProperty( "version", "0.1" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4469AuthenticatedDeploymentToCustomRepoTest.java:        verifier.setSystemProperty( "repositoryId", "mng4469" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4469AuthenticatedDeploymentToCustomRepoTest.java:        verifier.setSystemProperty( "repositoryUrl", "http://localhost:" + port + "/repo" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4590ImportedPomUsesSystemPropertiesTest.java:        verifier.setSystemProperty( "test.dir", testDir.getAbsolutePath() );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4615ValidateRequiredPluginParameterTest.java:        verifier.setSystemProperty( "config.requiredParam", "CLI" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4625SettingsXmlInterpolationWithXmlMarkupTest.java:            verifier.setSystemProperty( "test.prop", "\"&x=y<>\"" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4625SettingsXmlInterpolationWithXmlMarkupTest.java:            verifier.setSystemProperty( "test.prop", "&x=y<>" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng4765LocalPomProjectBuilderTest.java:        verifier.setSystemProperty( "user.prop", "OK" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5011ConfigureCollectionArrayFromSystemPropTest.java:        verifier.setSystemProperty( "config.stringParams", "" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5011ConfigureCollectionArrayFromSystemPropTest.java:        verifier.setSystemProperty( "config.fileParams", "foo,bar" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5011ConfigureCollectionArrayFromSystemPropTest.java:        verifier.setSystemProperty( "config.listParam", ",two,,four," );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng5208EventSpyParallelTest.java:        verifier.setSystemProperty( "maven.ext.class.path", "../spy/target/event-spy-0.0.1-SNAPSHOT.jar" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng0294MergeGlobalAndUserSettingsTest.java:            verifier.setSystemProperty( "org.apache.maven.global-settings", "global-settings.xml" );
core-it-suite/src/test/java/org/apache/maven/it/MavenITmng6558ToolchainsBuildingEventTest.java:        verifier.setSystemProperty( "maven.ext.class.path", "spy-0.1.jar" );
```

**Comment by jira-importer** [comment]
**[Slawomir Jaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slawekjaranowski)** commented

This is to resolve: MSHARED-1132

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [968c1a25f5535c7f53e273ec1b9ad499fd4f1dee](https://gitbox.apache.org/repos/asf?p=maven.git&a=commit&h=968c1a25f5535c7f53e273ec1b9ad499fd4f1dee), with [2dbee43408d4473dd5cfacff254ab2a592dde9db](https://gitbox.apache.org/repos/asf?p=maven.git&a=commit&h=2dbee43408d4473dd5cfacff254ab2a592dde9db) for maven-3.9.x branch, with with [7db942b2f7387f783d4530e2ee63897c204dd904](https://gitbox.apache.org/repos/asf?p=maven.git&a=commit&h=7db942b2f7387f783d4530e2ee63897c204dd904) for maven-3.8.x branch, ITs with [6906ad0704dd3e0d7a29b70a03640b3523eb59b7](https://gitbox.apache.org/repos/asf?p=maven-integration-testing.git&a=commit&h=6906ad0704dd3e0d7a29b70a03640b3523eb59b7), ITs with [4f4f8439f2d1e9967377feb3a463aa59b1fb4e8c](https://gitbox.apache.org/repos/asf?p=maven-integration-testing.git&a=commit&h=4f4f8439f2d1e9967377feb3a463aa59b1fb4e8c) for maven-3.8.x branch.
